Modules/star_system.py

```diff
@@ -0,0 +1,36 @@
+"""
+star_system.py - Represents a single star system.
+
+Copyright (c) 2018 The Fuel Rat Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+"""
+
+from dataclasses import dataclass
+from typing import Dict
+
+from utils.ratlib import Vector
+
+
+@dataclass(eq=True, frozen=True)
+class StarSystem:
+    position: Vector
+    name: str
+    spectral_class: str
+    is_populated: bool
+
+    @classmethod
+    def from_dict(cls, data: Dict):
+        return cls(Vector(data.get('x'),
```

Comment:
method should fail if any `data` fields are not defined.

Reply:
Fixed, with the exception of `spectral_class`. Not all systems have information on star class, so `None` is a reasonable value.